### ISSUE TITLE
Compatibility with Django 1.11

### ISSUE THREAD

**Issue by MacLake** [aldryn/aldryn-newsblog#440]
As Django CMS is now compatible with Django 1.10 and the support for 1.9 ends in April, it’s time to make newsblog also fit for 1.10.

One import error is caused by trying to import the obsolete patterns in urls.py, but after removing this, there’s something else going wrong.

**Comment by MacLake** [comment]
Now Django 1.9 is EOL, and got a last security update today.

I would help to fix the compatibility issue, but don’t know how. Can the creators of this app give a hint? Is it still maintained?

**Comment by evildmp** [comment]
@MacLake Yes it is most definitely still maintained. We'd love some help keeping it up-to-date. 

Can you send a message to the [django CMS Developers list](https://groups.google.com/forum/#!topic/django-cms-developers/8841mspjU_I) expressing your interest?

**Comment by aubricus** [comment]
I've started a post here: https://groups.google.com/forum/#!topic/django-cms-developers/-dg6EtQJyCg

**Comment by b14ckkn19ht** [comment]
Actually 1.10 isn't enough 1.11.3 or 4 should be the target in order to keep up...
I wonder what the necessary changes might be.

**Comment by LRH539** [comment]
The fix has apparently been merged with master, but pypi still hosts an outdated version which is incompatible with 1.10.

**Comment by artforlife** [comment]
@LRH539 does it mean this plugin is compatible with Django 1.11?

**Comment by LRH539** [comment]
@artforlife I'm running this plugin with Django 1.10, I think the devs are still working on Django 1.11 support, but 1.10 is acceptable for me. The version from pypi is still 1.3.3, but the master branch from this repository can be used instead:

```
pip install git+https://github.com/aldryn/aldryn-newsblog.git@master
```

You could try using Django 1.11, but I don't think that works (yet).

**Comment by artforlife** [comment]
@LRH539 Did you build a site with this setup from scratch or did you upgrade an existing site? I am asking because I have a site that is running Django 1.9 and the older version of this plugin. I am wondering what would happen if I upgraded from repository, as you suggest.

**Comment by LRH539** [comment]
@artforlife I've build everything from scratch. I've not used this plugin before. (I'm not using it in production yet, because I'm still facing some bugs). I suggest not upgrading until the devs officially released 1.4 to pypi, because it's not even stable when installing from scratch.

I'm mostly dealing with these issues: https://github.com/aldryn/aldryn-newsblog/issues/422, https://github.com/aldryn/aldryn-newsblog/issues/470 and https://github.com/aldryn/aldryn-newsblog/issues/471

**Comment by artforlife** [comment]
@LRH539 I see. Have you, per chance, encountered an issue where the NewsBlog works as intended when a user is logged in, but only displays the list of posts for anonymous users? That is, if it is an anonymous user, then even if he click to see the individual blog post, he still sees only the list of blog posts instead of specific content. However, the header and the lead-in text are displayed appropriately. I am not sure whether this is something known or perhaps something wrong with my setup.

**Comment by LRH539** [comment]
@artforlife I think this is exactly the bug I'm facing in #471, but I've misidentified the issue there. I've reopened a more specific issue #472, which describes exactly the behavior you are experiencing and I'm not sure what's going wrong. If you find a solution, please let me know, because this is currently the only reason I can't deploy.

**Comment by artforlife** [comment]
@LRH539 Do you experience this problem only with anonymous users or all users?

**Comment by LRH539** [comment]
This problem occurs with all users, we should move the discussion of that problem to #472 .

**Comment by artforlife** [comment]
@LRH539 Yes. But on the note of Django 1.10 compatibility, how did you deal with the problem described in #444 ?

**Comment by GabrielDumbrava** [comment]
@LRH539 solution also worked for me. Thanks a lot. I was experiencing bug #444. I think that [this](https://github.com/alexcustos/aldryn-newsblog/commit/c0a7cd1d38ddb0e8c51a79701771b7ec0215d831) commit may have also fixed this bug.

**Comment by LRH539** [comment]
@artforlife Yes, installing from master (as described above) fixed this bug for me as well.

**Comment by raphaa** [comment]
When is an update to pypi planned? Would like to use Django 1.10 (or better 1.11) without using the master branch :(

**Comment by LRH539** [comment]
I'm also waiting for an update, but I am now using the master branch in production without any problems so far. I must admit, that my site doesn't mainly rely on aldryn-newsblog. I only use it for a news section, but that works without any major problems, besides some bugs I experienced during the initial setup.

I posted my workarounds in the issues related to these bugs and they've worked on two sites so far. I suggest you shouldn't hesitate to use the master branch, because using an outdated version of Django is even worse than relying on this repo.
